### PR TITLE
Add reducers and actions

### PR DESCRIPTION
**Pull request by jonas-45**
In this milestone of the Bookstore project, react redux and redux toolkit were 
added with the following implementations;

- [ ] React store was added to the project for the state management.
- [ ] Addbook and Removebook reducers were added.
- [ ] A categories slice was also added with a checkStatus reducer implemented in it.

I however faced Missing file extension for "./features/books/booksSlice" and Missing file extension for "./categories/categoriesSlice" errors hence I added a rule to the eslint config file to disable only this error.

### CODE REVIEW

File: src/features/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { createSlice } from '@reduxjs/toolkit';
```
Comment (topeogunleye) [inline src/features/categories/categoriesSlice.js]: - [ ] According to project requirements, the structure of the files should follow the `ducks pattern`, please kindly move categoriesSlice.js to `/src/redux/categories/categoriesSlice.js`

File: src/features/books/booksSlice.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  books: [],
+};
+
+const booksSlice = createSlice({
```
Comment (topeogunleye) [inline src/features/books/booksSlice.js]: - [ ] According to project requirements, the structure of the files should follow the `ducks pattern`, please kindly move booksSlice.js to `/src/redux/books/booksSlice.js`

**Review by topeogunleye — APPROVED** [review]
### **STATUS: APPROVED  ✔️** 
------
Hi @jonas-45,  

Your project is complete! Great job for nailing it right on the first attempt 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟

wiserhood-cheers

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Followed Github flow :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good description :heavy_check_mark:

## [Optional] suggestions:
 * Nothing to mention.

 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear
Please, remember to tag @topeogunleye in your question so I can receive the notification

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by jonas-45** [comment]
Thanks @topeogunleye for your time and effort put into reviewing my work. All recommendations will be implemented right away.

**Pull request merged**